launch/launch/substitutions/boolean_substitution.py

```diff
@@ -172,21 +172,26 @@ class AnySubstitution(Substitution):
     If none of the arguments evaluate to true, then this substitution returns the string 'false'.
     """
 
-    def __init__(self, *args: SomeSubstitutionsType) -> None:
+    def __init__(
+        self, *args: SomeSubstitutionsType, container: List[SomeSubstitutionsType] | None = None,
+    ) -> None:
         """
         Create an AnySubstitution substitution.
 
         The following string arguments evaluate to true: '1', 'true', 'True', 'on'
         """
         super().__init__()
 
-        self.__args = [normalize_to_list_of_substitutions(arg) for arg in args]
+        if container is None:
+            container = []
+
+        self.__args = [normalize_to_list_of_substitutions(arg) for arg in chain(container, args)]
```

Comment:
A bit pedantic, but I would flip the order just to match the order of the constructor params, even if it shouldn't affect the any/all result:

```suggestion
        self.__args = [normalize_to_list_of_substitutions(arg) for arg in chain(args, container)]
```